source/vibe/data/json.d

```diff
@@ -69,20 +68,46 @@ import std.traits;
 	a JSONException. Additionally, an explicit cast or using get!() or to!() is
 	required to convert a JSON value to the corresponding static D type.
 */
+
+version (VibeJsonDisableBigInt)
+{
+	private enum BigIntSupport = false;
+}
+else
+{
+	import std.bigint;
+	version = VibeUseJsonBigInt;
+	private enum BigIntSupport = true;
+}
+
 struct Json {
 	private {
 		// putting all fields in a union results in many false pointers leading to
 		// memory leaks and, worse, std.algorithm.swap triggering an assertion
 		// because of internal pointers. This crude workaround seems to fix
 		// the issues.
-		void*[2] m_data;
-		ref inout(T) getDataAs(T)() inout { static assert(T.sizeof <= m_data.sizeof); return *cast(inout(T)*)m_data.ptr; }
-		@property ref inout(long) m_int() inout { return getDataAs!long(); }
-		@property ref inout(double) m_float() inout { return getDataAs!double(); }
-		@property ref inout(bool) m_bool() inout { return getDataAs!bool(); }
-		@property ref inout(string) m_string() inout { return getDataAs!string(); }
-		@property ref inout(Json[string]) m_object() inout { return getDataAs!(Json[string])(); }
-		@property ref inout(Json[]) m_array() inout { return getDataAs!(Json[])(); }
+		union Payload
+		{
+			version (VibeUseJsonBigInt)
+				BigInt bigint;
+			long long_;
+			double double_;
+			bool bool_;
+			string string_;
+			Json[string] object;
+			Json[] array;
+		}
```

Comment:
I missed this comment, sorry. And I don't understand it. Do you tell about replacing old void*[2] payload with this union?
We don't know the BigInt size (and we can't be sure, that it will not be changed in future), thus a union, I think, is the safest choise.